packages/module-service-discovery/src/client.ts

```diff
@@ -28,12 +31,12 @@ export class ServiceDiscoveryClient<T extends Environment = Environment>
     public endpoint: string;
     public http: IHttpClient;
 
-    get environment(): Promise<Environment> {
-        return firstValueFrom(
-            Query.extractQueryValue(
-                this.#query.query(undefined, { cache: { suppressInvalid: true } })
-            )
-        );
+    get environment(): T {
```

Comment:
Is it guaranteed you will find it in the queryCache? Should there be a fallback to the old method of fetching it?

Reply:
the old method is async and cannot be returned, hence the error below.
i assume this attribute is only accessed internally, but release of this change will force a major version